src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -60,86 +110,97 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
     public function configureOptions(OptionsResolver $resolver)
     {
-        $choiceListCache = &$this->choiceListCache;
         $registry = $this->registry;
-        $propertyAccessor = $this->propertyAccessor;
+        $choiceListFactory = $this->choiceListFactory;
+        $idReaders = &$this->idReaders;
+        $choiceLoaders = &$this->choiceLoaders;
         $type = $this;
 
-        $loader = function (Options $options) use ($type) {
-            $queryBuilder = (null !== $options['query_builder'])
-                ? $options['query_builder']
-                : $options['em']->getRepository($options['class'])->createQueryBuilder('e');
+        $choiceLoader = function (Options $options) use ($choiceListFactory, &$choiceLoaders, $type) {
+            // This closure and the "query_builder" options should be pushed to
+            // EntityType in Symfony 3.0 as they are specific to the ORM
```

Comment:
but the DoctrineType is the class holding the logic being shared by all Doctrine O*M projects. If it is specific to the ORM, it must go in the EntityType